packages/reporting-server/README.md

```diff
@@ -88,8 +88,148 @@ postgres://<user>:<password>@<host>/<database>
+
+We can create new user with the following command
+
+``` bash
+create user <youruser> with encrypted password <yourpassword>;
+
+# E.G: create user myuser with encrypted password 'mypassword';
+```
+
+After the user is created we need to assign the db roles to the user:
+
+``` bash
+grant all privileges on database <yourdatabase> to <youruser>;
+
+# E.G: grant all privileges on database reporting to myuser;
+```
+
+### Set/change user PostgreSQL password
+
+``` bash
+sudo -i -u postgres
+```
+
+Then you can access the Postgres prompt by typing:
+
+``` bash
+psql
+```
+
+To change your password you run the following command:
+
+``` bash
+\password <user>
+# E.g: \password postgres
+```
+
+To exit out of the PostgreSQL prompt, run the following:
+
+``` bash
+\q
+```
+
+
+
+### PostgresSQL URL
+
+Once the above steps are finished you can build your bd_url:
+
+``` 
+postgres://<user>:<password>@<host>/<database>
+```
+
+For more information you can check [this tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-postgresql-on-ubuntu-20-04-quickstart)
+
+
+## Run the application
+
+Now that you have already elected, configured and created a database and set you configurations. You need to run the migrations to create the database tables. We can proceed to apply migrations to create the database tables by running
+
+```bash
+aerich upgrade
+```
+
+After running this command your database should have all the tables defined in the code. Now you can run the application
+
+```bash
+reporting_server
+```
+
+If you want to run the migrations and start the server you could also run:
+
+```bash
+npm run start:dev
+```
+
 # Developers
 
+## Migration
+We are using [aerich](https://github.com/tortoise/aerich) as our database migration tool. That means that the changes to made to the model will not be reflected in the database automatically. You must run the aerich migration command to create a new migration with the changes:
+
+``` bash
+npm run migrate
+```
+
+If there are some changes in the Tortoise models, the previous command will create a migration (sql) file that has your changes. To apply these changes to the database you need to run
+
+``` bash
+npm run apply:migrations 
+```
+
+![aerich](https://user-images.githubusercontent.com/11761240/122826198-d97f2e80-d2b0-11eb-813f-384f4ae61d6b.png)
+
+To change the settings for the url, user and password you can open the [default_config file](https://github.com/open-rmf/rmf-web/blob/main/packages/reporting-server/rest_server/default_config.py) and change the value of the `db_url` with the following format (E.g PostgreSQL):
+
+```bash
+postgres://<user>:<password>@<host>/<database>
+```
+
+You can also add new Tortoise's model folders and other databases on the file `rmf-web/packages/reporting-server/rest_server/aerich_config.py`. You can read more about the configuration of Aerich [here](https://tortoise-orm.readthedocs.io/en/latest/migration.html)
```

Comment:
Is there some scenarios where one would need to change the models?

Reply:
If we refer to the tortoise models that represent database tables, absolutely. We will always want to add attributes, change attribute names, etc. All of this is going to be reflected in migrations.

For example, I just made a table refactor to try to solve the database size issue.

Reply:
Are all these meant to be done by the sysadmin? This seems like a pretty complicated config, might be worth it to have docs showing a situation where you need to modify the tortoise models and explaining how to do it.

Reply:
Well, that's a valid point, I thought that a developer would be in charge of creating new reports and modify the database, there is a couple of things to consider here
Do we want the sysadmin to modify the database? If yes, most of the frameworks that work with ORMs have this flow + complexity. If the sysadmin wants to modify the database it should have basic knowledge of databases. One example where we want to modify a table would be the following: 

Let's say we just register the start time and the ETA of a task. But after some time we are able to get the real end time of the task. So we want to add the `end_time` attribute to the table. In the following PR, we should add that `end_time` attribute to the `task` model, generate the migration (sql file with the change) and apply the migrations to the database (run the generated sql file with the changes).

If we want to modify the tables the less number possible times I think we should just work with Postgres JSON fields, but it'll be more complicated, we wouldn't be able to use FK, Tortoise filters will be useless because we need to query a JSON field within a table and since the tortoise ORM does not support a JSON query syntax we have to implement the query syntax of each database on the code and that's when we hit that point where the ORM lose it essence, this will be a good scenario for MongoDB but we cannot use it. Other things to consider we empirically saw few time ago, is more expensive to work with JSON fields, the logs occupied more space and queries are more expensive as well due to the number of fields that we manage.

Reply:
I agree that these config should be managed by the developer, in that case, I think this part could be in a separate developer section/doc, or put in somewhere in the source code as comments/docstrings.